advanced-javascript/tests/arrays.test.js

```diff
@@ -15,46 +15,67 @@ chai.use(sinonChai);
 // hint 3. - if the function you're testing requires a * callback *, make sure you use a spy like sinon
 
 describe('Arrays', () => {
+  let myTestArray = [];
+
+  beforeEach(() => {
+    myTestArray = ['Fred', 'Barney', 'Wilma', 'Betty'];
+  });
+
+  afterEach(() => {
+    myTestArray = [];
+  });
+
   describe('`each`', () => {
     it('should be a function', () => {
       const each = arrayFunctions.each;
-      expect(each).to.be.a('string');
+      expect(each).to.be.a('function');
+    });
+    it('should call a callaback for every item in a given array', () => {
+      const spy = sinon.spy();
+      const expectedCallTimes = myTestArray.length;
+      arrayFunctions.each(myTestArray, spy);
+      expect(spy).to.have.callCount(expectedCallTimes);
     });
-    // begin here
   });
 
   describe('`map`', () => {
+    const map = arrayFunctions.map;
     it('should be a function', () => {
-      const map = arrayFunctions.map;
-      expect(map).to.be.an('object');
+      expect(map).to.be.an('function');
+    });
+    it('call a callback for every item in a given array', () => {
+      const cb = sinon.spy();
+      const expectedCallTimes = myTestArray.length;
+      arrayFunctions.map(myTestArray, cb);
```

Comment:
Since you already defined `map` above, you could just call `map(myTestArray, cb);`